test/CheckedC/generic-function/generic-func-call-info.c

```diff
@@ -0,0 +1,29 @@
+// Test to make sure that the generic function calls are correctly instantiated
+//
+// In CallExpr, the type of the DeclRefExpr should be completely instantiated,
+// by replacing all type variable types with specified type names in the list
+// of type names provided when calling generic function.
+//
+//
+// RUN: %clang_cc1 -ast-dump -fcheckedc-extension %s | FileCheck %s

```

Comment:
Please add test cases for the different forms of constructed types:
- Pointers to type variables.
- Function types that have type variables within them.
- Array types that use type variables.

Also please add a test case for the new error message that you have introduced.